deployment/common/view/nops.go

```diff
@@ -56,42 +57,57 @@ func GenerateNopsView(nodeIDs []string, oc deployment.OffchainClient) (map[strin
 	} else if err != nil {
 		return nv, fmt.Errorf("failed to get node info: %w", err)
 	}
-	for _, node := range nodes {
-		// get node info
-		nodeDetails, err := oc.GetNode(context.Background(), &nodev1.GetNodeRequest{Id: node.NodeID})
-		if err != nil {
-			return nv, errors.Wrapf(err, "failed to get node details from offchain client for node %s", node.Name)
-		}
-		if nodeDetails == nil || nodeDetails.Node == nil {
-			return nv, fmt.Errorf("failed to get node details from offchain client for node %s", node.Name)
+	nodesResp, err := oc.ListNodes(context.Background(), &nodev1.ListNodesRequest{
+		Filter: &nodev1.ListNodesRequest_Filter{
+			Ids: nodeIDs,
+		},
+	})
+	if err != nil {
+		return nv, fmt.Errorf("failed to list nodes from JD: %w", err)
+	}
+	details := func(nodeID string) *nodev1.Node {
+		// extract from the response
+		for _, node := range nodesResp.Nodes {
+			if node.Id == nodeID {
+				return node
+			}
 		}
-		nodeName := nodeDetails.Node.Name
+		return nil
+	}
+	jobspecs, err := approvedJobspecs(context.Background(), nodeIDs, oc)
+	if err != nil {
+		// best effort on job specs
+		fmt.Printf("WARNING: Failed to get approved jobspecs: %v\n", err)
```

Comment:
wouldn't it be better to pass a logger around and log it with it instead of `fmt`?